PowerEditor/src/WinControls/Grid/ShortcutMapper.cpp

```diff
@@ -124,123 +122,176 @@ void ShortcutMapper::initBabyGrid() {
 	_babygrid.setHighlightColorProtectNoFocus(RGB(230,194,190));
 }
 
+generic_string ShortcutMapper::getTextFromCombo(HWND hCombo)
+{
+	TCHAR str[128];
+	::SendMessage(hCombo, WM_GETTEXT, 127, reinterpret_cast<LPARAM>(str));
+	generic_string res(str);
+	return stringToLower(res);
+};
+
+bool ShortcutMapper::isFilterValid(Shortcut sc)
+{
+	bool match;
+	generic_string shortcut_name = stringToLower(generic_string(sc.getName()));
+	match = false;
+	if (_shortcutFilter[0] == TEXT('\0')){
+		match = true;
+	}
+	// test the filter on the shortcut name
+	size_t match_pos = shortcut_name.find(_shortcutFilter);
+	if (match_pos != std::string::npos){
+		match = true;
+	}
+	return match;
+}
+
 void ShortcutMapper::fillOutBabyGrid()
 {
 	NppParameters *nppParam = NppParameters::getInstance();
 	_babygrid.clear();
 	_babygrid.setInitialContent(true);
+	_shortcutIndex.clear();
 
 	size_t nbItems = 0;
 
 	_babygrid.setText(0, 1, TEXT("Name"));
 	_babygrid.setText(0, 2, TEXT("Shortcut"));
-	
-	switch(_currentState) {
-		case STATE_MENU: {
-			nbItems = nppParam->getUserShortcuts().size();
-			_babygrid.setLineColNumber(nbItems, 3);
-			_babygrid.setText(0, 3, TEXT("Category"));
-			break; }
-		case STATE_MACRO: {
-			nbItems = nppParam->getMacroList().size();
-			_babygrid.setLineColNumber(nbItems, 2);
-			break; }
-		case STATE_USER: {
-			nbItems = nppParam->getUserCommandList().size();
-			_babygrid.setLineColNumber(nbItems, 2);
-			break; }
-		case STATE_PLUGIN: {
-			nbItems = nppParam->getPluginCommandList().size();
-			_babygrid.setLineColNumber(nbItems, 3);
-			_babygrid.setText(0, 3, TEXT("Plugin"));
-			break; }
-		case STATE_SCINTILLA: {
-			nbItems = nppParam->getScintillaKeyList().size();
-			_babygrid.setLineColNumber(nbItems, 2);
-			break; }
+
+	switch (_currentState) {
+	case STATE_MENU: {
+		nbItems = nppParam->getUserShortcuts().size();
+		_babygrid.setLineColNumber(nbItems, 3);
+		_babygrid.setText(0, 3, TEXT("Category"));
+		break; }
+	case STATE_MACRO: {
+		nbItems = nppParam->getMacroList().size();
+		_babygrid.setLineColNumber(nbItems, 2);
+		break; }
+	case STATE_USER: {
+		nbItems = nppParam->getUserCommandList().size();
+		_babygrid.setLineColNumber(nbItems, 2);
+		break; }
+	case STATE_PLUGIN: {
+		nbItems = nppParam->getPluginCommandList().size();
+		_babygrid.setLineColNumber(nbItems, 3);
+		_babygrid.setText(0, 3, TEXT("Plugin"));
+		break; }
+	case STATE_SCINTILLA: {

```

Comment:
This change of indentation doesn't belong in this commit. Plus, it makes the code layout inconsistent. See the `switch` statement immediately below, which also doesn't have a space.

Reply:
done